### ISSUE TITLE
aws_codebuild_project: add registryCredential for external Docker registry

### ISSUE THREAD

**Issue by srubasheuski** [terraform-providers/terraform-provider-aws#9147]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

For now we can specify link for the external docker registry, but we can't specify registry credentials.

### New or Affected Resource(s)

* aws_codebuild_project

### Potential Terraform Configuration

```hcl
resource "aws_codebuild_project" "test" {
  ...

  environment {
    compute_type                = "BUILD_GENERAL1_SMALL"
    image                       = "docker_registry.com/image:1.0.0"
    type                        = "LINUX_CONTAINER"
    image_pull_credentials_type = "SERVICE_ROLE"
    
    registry_credentials {
      credential_provider = "SECRET_MANAGER"
      credential          = "some_arn"
    }
  }

  ...
}
```

### References

* https://docs.aws.amazon.com/codebuild/latest/APIReference/API_RegistryCredential.html
* https://github.com/terraform-providers/terraform-provider-aws/issues/8028

**Comment by bflad** [comment]
Support for a new `registry_credential` configuration block inside the `environment` configuration block has been merged and will release with version 2.19.0 of the Terraform AWS Provider, likely later this week. Thanks to @iliazlobin!

**Comment by srubasheuski** [comment]
Thanks a lot!!!!

**Comment by bflad** [comment]
This has been released in [version 2.19.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2190-july-11-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!
